### ISSUE TITLE
Add spacing to `TextureAtlasBuilder`

### ISSUE THREAD

**Issue by st0rmbtw** [bevyengine/bevy#9986]
## What problem does this solve or what need does it fill?

It would be nice if `TextureAtlasBuilder` supported spacing from the top and left sides of a texture atlas.

## What solution would you like?

Add the spacing property to `TextureAtlasBuilder`.

**Comment by rparrett** [comment]
Just to clarify, you're looking for more control beyond what was added here, right?

https://github.com/bevyengine/bevy/pull/9494

**Comment by st0rmbtw** [comment]
@rparrett Yes, exactly

**Comment by superdump** [comment]
_Why_ would it be nice?

**Comment by st0rmbtw** [comment]
Sometimes you want to add not only a padding, but also a some offset from the top left corner for a texture atlas.
For example, in `bevy_ecs_tilemap` there is a `spacing` setting for a tilemap texture that includes both padding and offset from the top left corner.

**Comment by ickshonpe** [comment]
I'm still not really clear on what you want or why this is useful.
Do you mean that you want support for padding the top and left edges of the texture atlas with empty space? Or is it about translating all the texture rects by some offset?

**Comment by st0rmbtw** [comment]
![img](https://github.com/bevyengine/bevy/assets/61053971/fae980f3-5e0d-4b5c-b772-f6ca197ceeb0)
Red line is spacing
Green line is padding

**Comment by rparrett** [comment]
> Do you mean that you want support for padding the top and left edges of the texture atlas with empty space

This, and I assume the goal is exporting texture atlases specifically for use with `bevy_ecs_tilemap`? `bevy_ecs_tilemap` only supports loading grids with padding on the top and left equal to the padding between tiles. https://github.com/StarArawn/bevy_ecs_tilemap/issues/353

Let me know if I'm wrong, @st0rmbtw.

It seems like `TextureAtlasBuilder` sort of coincidentally would work for building grid atlases when used with images of identical size.

**Comment by st0rmbtw** [comment]
Yeah, I wanted to build a big texture atlas from different texture atlases, since `bevy_ecs_tilemap` doesn't support multiple texture atlases yet. But in `TextureAtlasBuilder` there is only a way to set padding between tiles, but not from the top left corner that is required for `bevy_ecs_tilemap`. So I think adding a possibility to set an offset from the top left corner alongside with padding wouldn't be superfluous.

**Comment by ickshonpe** [comment]
Merging texture atlases is pretty easy.
I didn't upstream this to Bevy because I'm not sure how useful it is and the implementation isn't great as it requires the creation of intermediate texture atlases:
```rust
pub fn merge_atlases<'a, I>(
    image_assets: &mut Assets<Image>,
    atlas_assets: &mut Assets<TextureAtlas>,
    atlas_handles: I,
) -> Handle<TextureAtlas>
where
    I: IntoIterator<Item = &'a Handle<TextureAtlas>>,
{
    let mut builder = TextureAtlasBuilder::default();
    let atlases = atlas_handles
        .into_iter()
        .map(|h| atlas_assets.get(h).expect("Missing texture atlas"))
        .collect::<Vec<_>>();
    for atlas in &atlases {
        let image = image_assets
            .get(&atlas.texture)
            .expect("Missing atlas texture");
        builder.add_texture(atlas.texture.clone_weak(), image);
    }
    let mut meta_atlas = builder.finish(image_assets).unwrap();
    let meta_rects = take(&mut meta_atlas.textures);
    let meta_handles = take(&mut meta_atlas.texture_handles).unwrap();
    for atlas in &atlases {
        let meta_rect_index = meta_handles[&atlas.texture];
        let meta_rect = meta_rects[meta_rect_index];
        for &rect in atlas.textures.iter() {
            let meta_sub_rect = Rect {
                min: meta_rect.min + rect.min,
                max: meta_rect.min + rect.max,
            };
            meta_atlas.add_texture(meta_sub_rect);
        }
    }
    atlas_assets.add(meta_atlas)
}
```
New sprite indices are ordered respective of the order of the input texture atlases.
Haven't tested it with Bevy 0.11, but should work.

**Comment by ickshonpe** [comment]
Oh I remember, it doesn't work any more because texture handles were made pub(crate) in 0.11.
